account_check_printing_report_base/models/account_payment.py

```diff
@@ -37,7 +37,8 @@ def do_print_checks(self):
     @api.multi
     def post(self):
         res = super(AccountPayment, self).post()
-        if (self.journal_id.check_print_auto and
-                self.payment_method_id.code == 'check_printing'):
-            return self.do_print_checks()
+        for rec in self:
+            if (rec.journal_id.check_print_auto and
+                    rec.payment_method_id.code == 'check_printing'):
+                return rec.do_print_checks()
```

Comment:
The problem now is that doing a return breaks the loop, not completing the task. First question: `do_print_checks` admit several records? If so just do:
```python
res = super(...)
recs = self.filtered(lambda x: x.journal_id.check_print_auto and
                     x.payment_method_id.code == 'check_printing')
if recs:
    return recs.do_print_checks()
return res
```
If not, simply remove the `return` word.

Reply:
yes, do_print_checks admits multiple records but i found that it also has a return statement inside it, therefor the loop is going to be broken in there as well... 

I think a good solution might be removing the return word..  or maybe this should be checked out deeper. 

what do you think ?

Reply:
Fixed and tested. 

Thanks.